src/adv.py

```diff
@@ -33,19 +37,96 @@
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
 
+# Add items to rooms
+room['outside'].items = [Item('chicken', 'yummy'), Item('plate', 'holds food')]
+room['foyer'].items = [Item('candle', 'for light'), Item(
+    'picture', 'a picture of a farmer')]
+room['overlook'].items = [Item('pencil', 'to write')]
+room['treasure'].items = [Item('key', 'it opens something..')]
 #
 # Main
 #
 
 # Make a new player object that is currently in the 'outside' room.
 
-# Write a loop that:
-#
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
-# * Waits for user input and decides what to do.
-#
-# If the user enters a cardinal direction, attempt to move to the room there.
-# Print an error message if the movement isn't allowed.
-#
-# If the user enters "q", quit the game.
+
+def initialize_game():
+    user_name = input("Welcome, please enter your name...\n")
+    global player
+    player = Player(user_name, room['outside'])
+    run_game(f"Welcome {user_name}")
+
+
+def run_game(message=""):
+    os.system('cls')
+    os.system('clear')
+    print(player.__str__())
+    if message:
+        print(f"******{message}******")
+    user = input(
+        'Enter a cardinal direction. N, W, S, or E\nAlternatively, you can get/take or use followed by an item name\n')
+    handle_input(user)
+
+
+def handle_input(_input):
```

Comment:
great job setting all these variables up outside of the main loop. I love this kind of code.